lib/internal/http2/compat.js

```diff
@@ -486,4 +520,41 @@ function onServerStream(stream, headers, flags) {
   server.emit('request', request, response);
 }
 
+class Http2PushRequest {
```

Comment:
FWIW my thoughts about the push API and compat layer.

1. Both npm:http2 and npm:spdy expose a compatibility layer with added support for PUSH_PROMISE.

   Quote: https://www.npmjs.com/package/http2#api
   > The goal is the perfect API compatibility, with additional HTTP2 related extensions (like server push).

1. I renamed it to `createPushRequest` to match ["Push Request" in the RFC](http://httpwg.org/specs/rfc7540.html#PushRequests), meaning the PUSH_PROMISE frame. The HEADERS/DATA frames are collectively referred to as the "Push Response," whose API is exactly the same as a regular `Http2ServerResponse`.

1. As a user of the H2 APIs it's important for me to be able to send the push requests (PUSH_PROMISE) before the main response (HEADERS/DATA for the user's request), followed finally by the push responses (HEADERS/DATA for the PUSH_PROMISE). This is the optimal flow described in: https://en.wikipedia.org/wiki/HTTP/2_Server_Push